src/models/keyframesmodel.cpp

```diff
@@ -583,20 +581,102 @@ bool KeyframesModel::isKeyframe(int parameterIndex, int position)
     return false;
 }
 
+bool KeyframesModel::advancedKeyframesInUse()
+{
+    if (m_filter && m_metadata && m_filter->animateIn() <= 0 && m_filter->animateOut() <= 0)
+    for (int i = 0; i < m_metadata->keyframes()->parameterCount(); i++) {
+        if (m_filter->keyframeCount(m_metadata->keyframes()->parameter(i)->property()) > 0) {
+            return true;
+        }
+    }
+    return false;
+}
+
+void KeyframesModel::removeAdvancedKeyframes()
+{
+    if (m_filter && m_metadata && m_filter->animateIn() <= 0 && m_filter->animateOut() <= 0) {
+        for (int i = 0; i < m_metadata->keyframes()->parameterCount(); i++) {
+            QString name = m_metadata->keyframes()->parameter(i)->property();
+            if (m_filter->keyframeCount(name) > 0) {
+                m_filter->set(name, m_filter->get(name, 0));
+                for (auto& key : m_metadata->keyframes()->parameter(i)->gangedProperties()) {
+                    m_filter->set(key, m_filter->get(key, 0));
```

Comment:
Use values at current frame position instead of first frame?

Reply:
I am willing to change that. But I am not sure it if would be better. 

This function is called when:
* Advanced keyframes are in use
AND
* The user grabs the simple keyframes handle
THEN
The user is prompted to ask if they want to clear simple keyframes with an option to cancel.

Why did they grab the handle if they have already done work to set advanced keyframes? It was probably a mistake and they will choose cancel. If it was not a mistake, did they carefully position the cursor on the exact frame that has a value that they want? Maybe not. Maybe the cursor is sitting arbitrarily between two advanced keyframes on a value that they never selected.

The current code (using keyframe 0) at least sets the parameter value to the first value the user had selected.

I am open to changing it, just wanted to give more perspective.

Reply:
I guess this is fine as-is and somewhat ambiguous either way.